slurm/slurm.py

```diff
@@ -0,0 +1,108 @@
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+os.environ["TEST_ENVAR_OUTSIDE"] = 'defined'
+
+affinity_script = """
+import os
+affinity = os.sched_getaffinity(0)
+print("num_cores: " + str(len(affinity)))
+for aff in affinity:
+    print("affinity: " + str(aff))"""
+
+
+class SlurmTestBase(rfm.RunOnlyRegressionTest):
+    descr = "Slurm test: "
+    valid_systems = required
+    valid_prog_environs = required
+    time_limit = '10m'
+    num_tasks = 1
+    num_tasks_per_node = 1
+    num_cpus_per_task = 1
+
+
+@rfm.simple_test
+class SbatchCleanEnvTest(rfm.RunOnlyRegressionTest):
+    descr += "sbatch starts in a clean environment"
+    exe = 'print(os.getenv("TEST_ENVAR_OUTSIDE") is None)'
+    executable = f"python3 -c 'import os;{exe}'"
+
+    @sanity_function
+    def assert_env(self):
+        return sn.assert_found(r'^True$', self.stdout, self.descr)
+
+
+@rfm.simple_test
+class SbatchSrunCopyEnvTest(rfm.RunOnlyRegressionTest):
+    descr += "srun copies the sbatch job environment"
+    prerun_cmds = ['export TEST_ENVAR_INSIDE=defined']
+    exe = 'print(os.environ["TEST_ENVAR_INSIDE"] == "defined")'
+    executable = f"srun python3 -c 'import os;{exe}'"
+
+    @sanity_function
+    def assert_env(self):
+        return sn.assert_found(r'^True$', self.stdout, self.descr)
+
+
+@rfm.simple_test
+class SbatchEnforceBindingTest(rfm.RunOnlyRegressionTest):
```

Comment:
can we also check that is actually works (in the gpu case)?

Reply:
that's planned for another test